packages/tracing/src/browser/request.ts

```diff
@@ -209,19 +216,18 @@ export function xhrCallback(
   shouldCreateSpan: (url: string) => boolean,
   spans: Record<string, Span>,
 ): void {
-  if (!handlerData || !handlerData.xhr || !handlerData.xhr.__sentry_xhr__) {
+  const currentClientOptions = getCurrentHub()
+    .getClient()
+    ?.getOptions();
+  if (
+    !(currentClientOptions && hasTracingEnabled(currentClientOptions)) ||
+    !(handlerData.xhr && handlerData.xhr.__sentry_xhr__ && shouldCreateSpan(handlerData.xhr.__sentry_xhr__.url)) ||
+    handlerData.xhr.__sentry_own_request__
+  ) {
     return;
   }
 
   const xhr = handlerData.xhr.__sentry_xhr__;
-  if (!shouldCreateSpan(xhr.url)) {
-    return;
-  }
-
-  // We only capture complete, non-sentry requests
-  if (handlerData.xhr.__sentry_own_request__) {
-    return;
-  }
```

Comment:
Quick refresher, why is this removed?

Reply:
It's not:

https://github.com/getsentry/sentry-javascript/pull/2921/files#diff-eff874a474b55b5fed7cab490cf0f993R222-R226